### PR TITLE
add heterogeneous design

### PR DESCRIPTION
**Pull request by DanielZhangQD**
### What problem does this PR solve? 
Add new heterogeneous design
### Check List 

Tests 

 - Unit test
 - E2E test
 - Stability test
 - Manual test (add detailed scripts or steps below)
 - No code

Code changes

 - Has Go code change
 - Has CI related scripts change
 - Has Terraform scripts change

Side effects

 - Breaking backward compatibility

Related changes

 - Need to cherry-pick to the release branch
 - Need to update the documentation

### Does this PR introduce a user-facing change?:

```release-note
NONE
```

### CODE REVIEW

File: docs/design-proposals/2020-06-12-heterogeneous-design-for-tidb-cluister.md [hunk 1]
```diff
@@ -1,4 +1,4 @@
-# Heterogeneous design for TidbCluster
+# Deprecated - Heterogeneous design for TidbCluster
```
Comment (cofyc) [inline docs/design-proposals/2020-06-12-heterogeneous-design-for-tidb-cluister.md:1]:
```suggestion
# Heterogeneous design for TidbCluster (DEPRECATED)
```

Comment (cofyc) [reply docs/design-proposals/2020-06-12-heterogeneous-design-for-tidb-cluister.md:1]: can you rename this file to `2020-06-12-heterogeneous-design-for-tidb-cluster.md`, it has a typo in the file name

Comment (DanielZhangQD) [reply docs/design-proposals/2020-06-12-heterogeneous-design-for-tidb-cluister.md]: done

**Comment by DanielZhangQD** [comment]
@cofyc PTAL again.

**Pull request merged**